framework/src/relationshipmanagers/GhostHigherDLowerDPointNeighbors.C

```diff
@@ -0,0 +1,96 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// App includes
+#include "GhostHigherDLowerDPointNeighbors.h"
+#include "Executioner.h"
+#include "FEProblemBase.h"
+#include "MooseApp.h"
+#include "GhostLowerDElems.h"
+
+// libMesh includes
+#include "libmesh/elem.h"
+#include "libmesh/mesh_base.h"
+#include "libmesh/boundary_info.h"
+
+registerMooseObject("MooseApp", GhostHigherDLowerDPointNeighbors);
+
+using namespace libMesh;
+
+InputParameters
+GhostHigherDLowerDPointNeighbors::validParams()
+{
+  InputParameters params = RelationshipManager::validParams();
+  params.set<bool>("attach_geometric_early") = false;
+  return params;
+}
+
+GhostHigherDLowerDPointNeighbors::GhostHigherDLowerDPointNeighbors(const InputParameters & params)
+  : RelationshipManager(params)
+{
+}
+
+GhostHigherDLowerDPointNeighbors::GhostHigherDLowerDPointNeighbors(
+    const GhostHigherDLowerDPointNeighbors & other)
+  : RelationshipManager(other)
+{
+}
+
+std::string
+GhostHigherDLowerDPointNeighbors::getInfo() const
+{
+  std::ostringstream oss;
+  oss << "GhostHigherDLowerDPointNeighbors";
+  return oss.str();
+}
+
+void
+GhostHigherDLowerDPointNeighbors::operator()(const MeshBase::const_element_iterator & range_begin,
+                                             const MeshBase::const_element_iterator & range_end,
+                                             const processor_id_type p,
+                                             map_type & coupled_elements)
+{
+  mooseAssert(_moose_mesh,
+              "The MOOSE mesh must be non-null in order for this relationship manager to work.");
+  const auto & node_to_elem_map = _moose_mesh->nodeToElemMap();
```

Comment:
How do we know in this map whether it's a lowerD-element or higherD-element node neighbor? (It's probably the wrong question, but it's showing what I don't understand about the coupled ghosting objects).

Reply:
we don't know *a priori*. The map will map from node to both higher and lower-d elements that use that node. That's why later in the code I check the dimension of the elements that we get from the node key

Reply:
Got it.